File: javascript/createProcess.js
const { createProcess, uploadBpmnFile } = require("./lib/ProcessMindApi.js");

async function main() {
	const config = {
		apiKey: "",      // Your API key
		tenantId: "",    // Your tenant ID
	};

	// Create a new process
	const process = await createProcess({
		...config,
		displayName: "My New Process"
	});
	console.log("Created process:", process);

	// Optionally upload a BPMN file to the process
	// await uploadBpmnFile({
	// 	...config,
	// 	processId: process.id,
	// 	filePath: "./my-process.bpmn"
	// });
	// console.log("BPMN uploaded successfully");
}

main().catch(console.error);

File: python/createProcess.py
"""
Create a new process in ProcessMind.

Configuration:
- Update api_key and tenant_id with your values.
"""

import requests

# === Configuration ===
api_key = ""        # Your API key
tenant_id = ""      # Your tenant ID
display_name = "My New Process"

api_url = "https://api.processmind.com"

print("Creating process...")
response = requests.post(
    f"{api_url}/tenant/{tenant_id}/processes",
    headers={"x-api-key": api_key, "Content-Type": "application/json"},
    json={"displayName": display_name, "treeNodeType": "Process"}
)

if not response.ok:
    print(f"Error: {response.status_code}, {response.reason}: {response.text}")
    exit(1)

process = response.json()
print(f"Process created: {process}")
